src/pages/Login.js

```diff
@@ -0,0 +1,138 @@
+import React, { useState, useEffect } from "react";
+import { auth, db } from "../firebase";
+import { ref, get } from "firebase/database";
+import { AlertError } from "../components/Alerts";
+import { signInWithEmailAndPassword } from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+import { TextboxWithLabels } from "../components/Textbox";
+
+const Login = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [message, setMessage] = useState("");
+  const [response, setResponse] = useState("");
+  const [showErrorAlert, setShowErrorAlert] = useState(false);
+
+  const navigate = useNavigate();
+
+  const handleSignin = async (e) => {
+    e.preventDefault();
+    if (!email || !password) return;
+
+    try {
+      const data = await signInWithEmailAndPassword(auth, email, password);
+      const uid = data.user.uid;
+
+      checkStudentDBV2(uid);
+      checkTeacherDBV2(uid);
```

Comment:
If it isn't a student, wouldn't it have to be a teacher? Can a user have both roles? If it is the former scenario, you could just assume it is a teacher once the student check fails instead of requesting data from the db twice. I think overall, splitting the data in the DB into teacher and student is not very easy to handle. It would be better to have a "User", which has a `role` property instead.